lib/sum_of_depository_balances_latest/main.py

```diff
@@ -27,40 +27,50 @@ async def get_sum_of_depository_balances_latest(
         The latest balance summed across all depository accounts
     """
 
+    # STEP 1: get a list of all institutions for the user
     institutions = await api_client.institutions.get(user_uuid=user_uuid)
     utc_starttime = utc_time - cutoff_interval
 
     # subset to only fetch data for institutions known to contain depository-type accounts for the user
-    institutions_w_depository = [inst for inst in institutions if "depository" in inst.account_types]
-
-    inst_coroutines = [
-        api_client.balances.get(
-            user_uuid=user_uuid,
-            institution_id=institution.institution_id,
-            utc_starttime=utc_starttime,
-            utc_endtime=utc_time,
-            account_types=["depository"],
+    institutions_w_depository = []
+    for inst in institutions:
+        if "depository" in inst.account_types:
+            institutions_w_depository.append(inst)
+
+    # STEP 2: get a list of all balances for each institution
+    inst_coroutines = []
+    for inst in institutions_w_depository:
+        inst_coroutines.append(
+            api_client.balances.get(
+                user_uuid=user_uuid,
+                institution_id=inst.institution_id,
+                utc_starttime=utc_starttime,
+                utc_endtime=utc_time,
+                account_types=["depository"],
         )
-        for institution in institutions_w_depository
-    ]
+    )
 
-    r = await asyncio.gather(*inst_coroutines)
-    records_list = []
-    for ix, inst_list in enumerate(r):
+    balances_per_institution = await asyncio.gather(*inst_coroutines)
+
+    # STEP 3: We flatten the lists of balances into a single list of balances
+    balances_flattened = []
+    for ix, inst_list in enumerate(balances_per_institution):
         institution_id = institutions[ix].institution_id
-        records_list.extend([dict(transaction, institution_id=institution_id) for transaction in inst_list])
+        for balance in inst_list:
+            balances_flattened.append(dict(balance, institution_id=institution_id))
     
-    # Here we sort by timestamp so latest balances are on top
-    records_list = sorted(records_list, key=lambda x: x["ts"], reverse=True)
+    # STEP 5: Here we sort by timestamp so latest balances are on top
+    balances_flattened = sorted(balances_flattened, key=lambda x: x["ts"], reverse=True)
```

Comment:
same comment as before. I can't think of a better way that's more readable. Can you or do you feel this is pretty straight forward?

Reply:
As I commented in previous PRs, I think most modern languages already have a lambda-like sort base function so this is not so just-python stuff. I will add this to potential final refactors.